src/compile/legend/rules.ts

```diff
@@ -21,7 +21,7 @@ export function values(legend: Legend) {
   if (vals && isDateTime(vals[0])) {
     return (vals as DateTime[]).map((dt) => {
       // normalize = true as end user won't put 0 = January
-      return timestamp(dt, true);
+      return {signal: dateTimeExpr(dt, true)};
```

Comment:
Actually Jeff already fix this.  Let's test this once new Vega version is released and merge then.